src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

```diff
@@ -999,26 +1015,30 @@ private function registerValidationConfiguration(array $config, ContainerBuilder
 
     private function getValidatorMappingFiles(ContainerBuilder $container, array &$files)
     {
-        if (interface_exists('Symfony\Component\Form\FormInterface')) {
-            $reflClass = new \ReflectionClass('Symfony\Component\Form\FormInterface');
+        $container->addResource(new ClassExistenceResource(Form::class));
+        if (class_exists(Form::class)) {
+            $reflClass = new \ReflectionClass(Form::class);
             $files['xml'][] = $file = dirname($reflClass->getFileName()).'/Resources/config/validation.xml';
             $container->addResource(new FileResource($file));
         }
 
         foreach ($container->getParameter('kernel.bundles_metadata') as $bundle) {
             $dirname = $bundle['path'];
 
-            if (is_file($file = $dirname.'/Resources/config/validation.yml')) {
+            $container->addResource(new FileExistenceResource($file = $dirname.'/Resources/config/validation.yml'));
+            if (is_file($file)) {
                 $files['yml'][] = $file;
                 $container->addResource(new FileResource($file));
             }
 
-            if (is_file($file = $dirname.'/Resources/config/validation.xml')) {
+            $container->addResource(new FileExistenceResource($file = $dirname.'/Resources/config/validation.xml'));
```

Comment:
We could optimize things by adding it only when the file does not exist (as we already have a FileResource when it exists, and removing the file will also invalidate the FileResource).
Or we should remove the FileResource if the content of the file does not impact the container itself (which may well be the case for this particular case)